src/libraries/Microsoft.PowerFx.Core/Binding/VoidToErrorTexlVisitor.cs

```diff
@@ -0,0 +1,138 @@
+
+namespace Microsoft.PowerFx.Core.Binding
+{
+    /// <summary>
+    /// Detects void expressions and converts them to error expressions.
+    /// </summary>
+    internal class VoidToErrorTexlVisitor : TexlVisitor
+    {
+        private readonly TexlBinding _txb;
+
+        public VoidToErrorTexlVisitor(TexlBinding txb)
+        {
+            _txb = txb ?? throw new ArgumentNullException(nameof(txb));
+        }
+
+        private void EnsureNotVoid(TexlNode node)
+        {
+            if (_txb.GetTypeAllowInvalid(node) == DType.Void)
+            {
+                _txb.ErrorContainer.EnsureError(node, TexlStrings.ErrBadType_NonBehavioralVoidExpression);
+            }
+        }
+
+        public override void PostVisit(StrInterpNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void PostVisit(DottedNameNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void PostVisit(UnaryOpNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void PostVisit(BinaryOpNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void PostVisit(VariadicOpNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void PostVisit(CallNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void PostVisit(ListNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void PostVisit(RecordNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void PostVisit(TableNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void PostVisit(AsNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(TypeLiteralNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(ErrorNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(BlankNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(BoolLitNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(StrLitNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(NumLitNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(DecLitNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(FirstNameNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(ParentNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
+        public override void Visit(SelfNode node)
+        {
+            EnsureNotVoid(node);
+        }
+
```

Comment:
Those nodes can never have void values; do we need those overrides?

Reply:
yes I believe most of them can never be void, keeping it just for future proofing...